src/groupByKey.ts

```diff
@@ -2,6 +2,19 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const grouped: GroupsMap<T> = {};
+
+  for (const item of items) {
+    const groupKey = item[key];
+
+    if (typeof groupKey === 'string' || typeof groupKey === 'number') {
```

Comment:
The type check for `groupKey` only allows strings and numbers. If `key` can be of other types, such as symbols or booleans, you might need to adjust this check or ensure that `key` is always a string or number in your use cases.